giskard/llm/client/utils.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import os
+from enum import Enum
+from typing import Optional
+
+
+class ApiType(Enum):
+    openai = (0,)
+    azure = 1
+
+
+def autodetect_client():
+    if "GISKARD_SCAN_LLM_API" in os.environ:
+        try:
+            return ApiType[os.environ["GISKARD_SCAN_API"].lower()]
```

Comment:
Mismatch in the env variable name. I would not link it to the scan only (note that we are not in the `scanner` module), because it’s also used in the tests.

`GSK_LLM_API`? (We use `GSK` instead of `GISKARD` for other env variables)

Reply:
As we do with other settings, I would put a global python variable somewhere instead of only relying on the environment variable. We should always allow to set things programmatically via Python.

```py3
import giskard as gsk

gsk.llm.set_llm_api(LLMClientAPI.OPEN_AI)
```

The environment should only be used to set some default.

I would put something like this in `__init__.py`:
```py3
_default_client_api = os.getenv("GSK_LLM_API", LLMClientAPI.AZURE if "AZURE_OPENAI_API_KEY" in os.environ else LLMClientAPI.OPEN_AI)
```